tests/suites/test_suite_bignum_mod.function

```diff
@@ -80,7 +80,178 @@ exit:
+                MBEDTLS_MPI_MOD_REP_MONTGOMERY ) );
+
+    TEST_EQUAL( 0, mbedtls_mpi_mod_residue_setup( &r, &m, R , n_limbs ) );
+
+    /* Pass for input_r < modulo */
+    TEST_EQUAL( 0, mbedtls_mpi_mod_read( &r, &m, r_buff, 1, endian ) );
 
+    /* Pass for input_r == modulo -1 */
+    memset( r_buff, 0xfd, buff_bytes );
+    TEST_EQUAL( 0, mbedtls_mpi_mod_read( &r, &m, r_buff, 1, endian ) );
+
+    /* modulo->p == NULL || residue->p == NULL ( m2 has not been set-up ) */
+    TEST_EQUAL( MBEDTLS_ERR_MPI_BAD_INPUT_DATA,
+                mbedtls_mpi_mod_read( &r, &m2, r_buff, 1, endian ) );
+    TEST_EQUAL( MBEDTLS_ERR_MPI_BAD_INPUT_DATA,
+                mbedtls_mpi_mod_read( &rn, &m, r_buff, 1, endian ) );
+    TEST_EQUAL( MBEDTLS_ERR_MPI_BAD_INPUT_DATA,
+                mbedtls_mpi_mod_write( &r, &m2, r_buff, 1, endian ) );
+    TEST_EQUAL( MBEDTLS_ERR_MPI_BAD_INPUT_DATA,
+                mbedtls_mpi_mod_write( &rn, &m, r_buff, 1, endian ) );
+
+    /* Fail for r_limbs < m->limbs */
+    r.limbs = m.limbs - 1;
+    TEST_EQUAL( MBEDTLS_ERR_MPI_BAD_INPUT_DATA,
+                mbedtls_mpi_mod_read( &r, &m, r_buff, 1, endian ) );
+    TEST_EQUAL( MBEDTLS_ERR_MPI_BAD_INPUT_DATA,
+                mbedtls_mpi_mod_write( &rn, &m, r_buff, 1, endian ) );
+    r.limbs = r_limbs;
+
+    /* Fail if input_r >= modulo m */
+    /* input_r = modulo */
+    memset( r_buff, 0xfe, buff_bytes );
+    TEST_EQUAL( MBEDTLS_ERR_MPI_BAD_INPUT_DATA,
+                mbedtls_mpi_mod_read( &r, &m, r_buff, 1, endian ) );
+
+    /* input_r > modulo */
+    memset( r_buff, 0xff, buff_bytes );
+    TEST_EQUAL( MBEDTLS_ERR_MPI_BAD_INPUT_DATA,
+                mbedtls_mpi_mod_read( &r, &m, r_buff, 1, endian ) );
+
+    /* Data too large to fit */
+    TEST_EQUAL( MBEDTLS_ERR_MPI_BUFFER_TOO_SMALL,
+                mbedtls_mpi_mod_read( &r, &m, r_buff, buff_bytes, endian ) );
+
+    /* Read the two limbs input data into a larger modulus and residue */
+    TEST_EQUAL( 0, mbedtls_mpi_mod_modulus_setup( &m2, N2, n2_limbs,
+                MBEDTLS_MPI_MOD_REP_MONTGOMERY ) );
+    rn.p = R2;
+    rn.limbs = r2_limbs;
+    TEST_EQUAL( MBEDTLS_ERR_MPI_BUFFER_TOO_SMALL,
+                mbedtls_mpi_mod_write( &rn, &m2, r_buff, 1, endian ) );
+
+exit:
+    mbedtls_mpi_mod_modulus_free( &m );
+    mbedtls_mpi_mod_modulus_free( &m2 );
+    mbedtls_free( N );
+    mbedtls_free( R );
+    mbedtls_free( N2 );
+    mbedtls_free( R2 );
+    mbedtls_free( r_buff );
+}
+/* END_CASE */
+
+/* BEGIN_CASE */
+void mpi_mod_io( char * input_N, data_t * input_A, int endian )
+{
+    mbedtls_mpi_uint *N = NULL;
+    mbedtls_mpi_uint *R = NULL;
+    unsigned char *r_buff = NULL;
+    mbedtls_mpi_mod_modulus m;
+    mbedtls_mpi_mod_residue r;
+    size_t n_limbs, n_bytes, a_bytes;
+
+    /* Read inputs */
+    TEST_EQUAL( 0, mbedtls_test_read_mpi_core( &N, &n_limbs, input_N ) );
+    n_bytes = n_limbs * sizeof( mbedtls_mpi_uint );
+    a_bytes = input_A->len * sizeof( char );
```

Comment:
Let's be consistent in size conversions (or lack of conversions).
```suggestion
    n_bytes = CHARS_TO_LIMBS( n_limbs );
    a_bytes = input_A->len;
```

Reply:
+1 to removing `sizeof(char)`, since that's 1 pretty much by definition.

However, a lot of the test code uses the `* xizeof( mbedtls_mpi_uint )` idiom, so I disagree that it needs to be changed here

Reply:
`CHARS_TO_LIMBS` won't work here, we want to convert the way around from limbs to chars and I think for that `n_limbs * sizeof( mbedtls_mpi_uint )` is correct.